### DISCUSSION TITLE
DigiKeyboard F20 and others

### DISCUSSION THREAD

**Discussion by tswfi** [ArminJo/DigistumpArduino#23] [Q&A] [answered]
Hi,

I'm trying to implement a mini keyboard (one key) that just sends F20 when pressed. I can send all the normal keys just fine and F11 works just fine but F20 does not.

Looking at the source I have a feeling this cuts F20 out. (didn't debug too deeply yet)

https://github.com/ArminJo/DigistumpArduino/blob/master/digistump-avr/libraries/DigisparkKeyboard/DigiKeyboard.h#L177

And also does this cap the keyboard to 101 keys?

https://github.com/ArminJo/DigistumpArduino/blob/master/digistump-avr/libraries/DigisparkKeyboard/DigiKeyboard.h#L49

Hardware: Chinese digispark clone (the one that plugs directly to usb port)

code 
```c
#define LAYOUT_FINNISH
#include "DigiKeyboard.h"

#define BUTTONPIN PB0

void setup() {
  pinMode(BUTTONPIN, INPUT_PULLUP);
  DigiKeyboard.enableLEDFeedback();
}

void loop() {
  if(digitalRead(BUTTONPIN) == LOW) {
    // this works (yes, really needs debounce :)
    // DigiKeyboard.sendKeyStroke(KEY_F11);

    // this does not work
    DigiKeyboard.sendKeyStroke(KEY_F20);
  }
  DigiKeyboard.delay(100);
}
```

Changing the LAYOUT define does not seem to have any effect on this.

And the reason for wanting to send F20 is that it is defined as mic mute by default on most linux distros :)

Any ideas?

**Comment by ArminJo** [comment]
> And also does this cap the keyboard to 101 keys?
> 
> https://github.com/ArminJo/DigistumpArduino/blob/master/digistump-avr/libraries/DigisparkKeyboard/DigiKeyboard.h#L49

Maybe, never tried to change it.
Please report if you have success.

**Comment by tswfi** [comment]
Ok, so changing the descriptor seems to be do the trick:

```
diff --git a/digistump-avr/libraries/DigisparkKeyboard/DigiKeyboard.h b/digistump-avr/libraries/DigisparkKeyboard/DigiKeyboard.h
index e42dd4d..32f0852 100644
--- a/digistump-avr/libraries/DigisparkKeyboard/DigiKeyboard.h
+++ b/digistump-avr/libraries/DigisparkKeyboard/DigiKeyboard.h
@@ -46,9 +46,9 @@ const PROGMEM uchar usbHidReportDescriptor[USB_CFG_HID_REPORT_DESCRIPTOR_LENGTH]
         0x81, 0x02,                    //   INPUT (Data,Var,Abs)
         0x95, 0x01,                    //   REPORT_COUNT (simultaneous keystrokes)
         0x75, 0x08,                    //   REPORT_SIZE (8)
-        0x25, 0x65,                    //   LOGICAL_MAXIMUM (101)
+        0x25, 0x73,                    //   LOGICAL_MAXIMUM (115)
         0x19, 0x00,                    //   USAGE_MINIMUM (Reserved (no event indicated))
-        0x29, 0x65,                    //   USAGE_MAXIMUM (Keyboard Application)
+        0x29, 0x73,                    //   USAGE_MAXIMUM (Keyboard Application)
         0x81, 0x00,                    //   INPUT (Data,Ary,Abs)
         0xc0                           // END_COLLECTION
         };
```

But I'm a bit perplexed as my PlatformIO has installed different version than is found on this repo. I will open a PR with that change but how does one tell platformio on vscode to get the platform from my forked repo and not the one that is shipped automatically?

**Comment by tswfi** [comment, marked as answer]
Opened a issue for this #24 and a pr that fixes it (at least for me) #25
